README.lhs

```diff
@@ -26,8 +26,6 @@ All built on the well-known `MonadLogger` interface and using an efficient
 ```haskell
 {-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DeriveGeneric #-}
```

Comment:
You should add `DerivingVia` here and not in `package.yaml`. The library doesn't need it, only the uses do.

You _could_ avoid messing with GHC-support on CI then, if you avoided running the `README` test conditionally, but I suppose there's not much point to verify Blammo builds on an older GHC if it can't be _used_ on an older GHC.